src/ssl_stubs.c

```diff
@@ -262,7 +263,55 @@ CAMLprim value ocaml_ssl_get_error_string(value unit)
   CAMLreturn(caml_copy_string(buf));
 }
 
+CAMLprim value ocaml_ssl_get_error_struct(value unit)
+{
+  CAMLparam1(unit);
+  unsigned long code = ERR_get_error();
+  CAMLlocal3(result, libstring, reasonstring);
```

Comment:
nit: swap to keep the 2 CAML-related lines together?

```suggestion
  CAMLlocal3(result, libstring, reasonstring);
  unsigned long code = ERR_get_error();
```